src/main/java/vic/cinema/service/impl/OrderServiceImpl.java

```diff
@@ -6,26 +6,30 @@
 import vic.cinema.lib.Inject;
 import vic.cinema.lib.Service;
 import vic.cinema.model.Order;
-import vic.cinema.model.Ticket;
+import vic.cinema.model.ShoppingCart;
 import vic.cinema.model.User;
 import vic.cinema.service.OrderService;
+import vic.cinema.service.ShoppingCartService;
 
 @Service
 public class OrderServiceImpl implements OrderService {
     @Inject
     private OrderDao orderDao;
+    @Inject
+    private ShoppingCartService shoppingCartService;
 
     @Override
     public List<Order> getOrderHistory(User user) {
         return orderDao.getAllByUser(user);
     }
 
     @Override
-    public Order completeOrder(List<Ticket> tickets, User user) {
+    public Order completeOrder(ShoppingCart cart) {
         Order order = new Order();
-        order.setTickets(tickets);
-        order.setUser(user);
+        order.setTickets(cart.getTickets());
```

Comment:
it is better to set a copy of tickets list here
I addressed this issue yesterday in Slack